projects/filjs-audit/README.md

```diff
@@ -0,0 +1,391 @@
+
+## Lotus RPC API
+
+### OpenRPC support
+> JSON based API definition of the lotus json-rpc interfaces that are auto-generated from the code.
+
+The Good: 
+- it is generated from the go rpc implementation, so is always up to date. 
+- It could be used to generate api docs and (almost complete) typescript and rust clients.
+
+The Bad: 
+- Not yet auto-discoverable or useable with existing tools like https://playground.open-rpc.org (the api definition doc should be discoverable from the api url, but currently isn't, probably due to auth, TBC)
+
+By manually extracting the definition files from the repo you can generate api docs from it:
+https://playground.open-rpc.org/?url=https://gist.githubusercontent.com/olizilla/fc6abf836023b6ee662955695f1cfd21/raw/a4cb5aa3c6fbbc47bcf4b92d2370b8b012a810cb/full.json
+
+API client generation is possible for Typescript and rust, but [it's rough](https://github.com/olizilla/lotus-openrpc-client/blob/main/client/typescript/src/index.ts#L280). This could be improved by adding more detail to the lotus openrpc definition (we should do that) but may also require improving the open-roc/generator project
+
+Notably, we cannot define the file import api with open-rpc, as it only specifies how to define JSON payloads, so we will always need a custom step to create a client or docs that includes the file import api.
+
+It could be used to simplify the custom generation code in https://github.com/filecoin-shipyard/js-lotus-client-schema
+
+Exploration: https://github.com/olizilla/lotus-openrpc-client
+
+Last updated: 31 March 2021 _(auto genereated from rpc code changes)_
+### lotus-gateway
+
+https://github.com/filecoin-project/lotus/tree/master/cmd/lotus-gateway
+
+## filecoin-shipyard JavaScript libraries
+
+### [js-lotus-client](https://github.com/filecoin-shipyard/js-lotus-client)
+
+Last updated: 20 Jan 2021
+
+#### https://github.com/filecoin-shipyard/js-lotus-client-rpc
+
+#### https://github.com/filecoin-shipyard/js-lotus-client-schema
+
+#### https://github.com/filecoin-shipyard/js-lotus-provider-browser
+
+#### https://github.com/filecoin-shipyard/js-lotus-provider-nodejs
+
+### [Filecoin.js](https://github.com/filecoin-shipyard/filecoin.js)
+
+#### Maturity
+
+The current state of the library specified in the README is `ALPHA: things will not work or work incorectly, will break and the API will change! ⚠️`
+
+Taking into account the activity of the repo, it has not been in actively development since November 2020 (Lotus version: ?), relying on an outdated fork of `filecoin-signing-tools-js`. There is a [Roadmap Issue](https://github.com/filecoin-shipyard/filecoin.js/issues/1) where we can see that general Filecoin operations are no yet implemented.
+
+When installing the module via npm latest, I got `"0.0.5-alpha"`even though github master `package.json` shows `"0.0.3-alpha"`.
```

Comment:
😱 might be some missing code we need to chase down here, will make a TODO for that